src/retry.decorator.ts

```diff
@@ -15,49 +15,58 @@ export function Retryable(options: RetryOptions): Function {
    * to the nature how arrow function defines this inside.
    *
    */
-  return function (target: Object, propertyKey: string, descriptor: TypedPropertyDescriptor<any>) {
+  return function(target: Object, propertyKey: string, descriptor: TypedPropertyDescriptor<any>) {
     const originalFn: Function = descriptor.value;
     // set default value for ExponentialBackOffPolicy
     if (options.backOffPolicy === BackOffPolicy.ExponentialBackOffPolicy) {
       !options.backOff && (options.backOff = 1000);
-      options.exponentialOption = { ...{ maxInterval: 2000, multiplier: 2 }, ...options.exponentialOption }
+      options.exponentialOption = {
+        ...{ maxInterval: 2000, multiplier: 2 },
+        ...options.exponentialOption
+      };
     }
-    descriptor.value = async function (...args: any[]) {
+    descriptor.value = async function(...args: any[]) {
       try {
-        return await retryAsync.apply(this, [originalFn, args, options.maxAttempts, options.backOff]);
+        return await retryAsync.apply(this, [originalFn, args, options.maxAttempts, options.backOff, options.doRetry]);
       } catch (e) {
-        e.message = `Failed for '${propertyKey}' for ${options.maxAttempts} times.`;
+        if (e.message === "maxAttempts") {
+          e.code = "429";
```

Comment:
any reason you want an error code `429` here?

Reply:
to continue handle this error it's more easy to compare by `code` and 429 (to many requests) is currently the most logic for me :)